src/search-manager/data/apiHooks.ts

```diff
@@ -221,3 +241,27 @@ export const useTagFilterOptions = (args: {
 
   return { ...mainQuery, data };
 };
+
+/* istanbul ignore next */
+export const useGetSingleDocument = ({ client, indexName, id }: {
+  client?: MeiliSearch;
+  indexName?: string;
+  id: string | number;
+}) => (
+  useQuery({
+    enabled: client !== undefined && indexName !== undefined,
+    queryKey: [
+      'content_search',
+      client?.config.apiKey,
+      client?.config.host,
+      indexName,
+      id,
+    ],
+    queryFn: () => {
+      if (client === undefined || indexName === undefined) {
+        throw new Error('Required data unexpectedly undefined. Check "enable" condition of useQuery.');
+      }
+      return fetchDocumentById({ client, indexName, id });
+    },
+  })
+);
```

Comment:
Since the meilisearch contains all data related to a collection, I wanted to fetch individual document from it. But meilisearch [getDocument](https://www.meilisearch.com/docs/reference/api/documents#get-one-document) method fails with invalid api-key if we use tenant key. Checked everything related to tenant keys, like allowed `actions` and `search_rules` but everything seems in order. It works with main api-key but not with tenant keys even if it has same permissions as the main key. 

**Current Solution:** Used collections api to fetch. Although not used, did not remove this api and hook just in case. Can remove this if we don't want to add unused functions.

Reply:
My bet is that the tenant key can only be used in the `search` endpoint. But, as you said, this is not documented anywhere. The only hint is that they only show examples for tenant keys with search.

Reply:
If we're just loading a single "thing", I think it's usually better to load it from the REST API anyways - it's guaranteed to be up to date, and we don't need any of the filtering/keyword/sorting features that Meilisearch provides. What's the exact use case here? Getting the collection metadata + tags in a single call instead of two calls?

Reply:
@bradenmacdonald Using Meilisearch here gets us the collection metadata including tags, plus its components in a single (multisearch) hit.

If we're going to start using the REST API in some places and the search index in others, I'd rather this was planned and done as part of a separate ticket. E.g. if we need to make upserting to Meilisearch async, but always want to be able to see the most recent changes made to a library/collection/component, then using the REST API to get after creating/updating makes sense. But while we're upserting synchronously, using the search index alone is fine.

Reply:
That makes sense to me.